python/run-tests.py

```diff
@@ -99,8 +99,60 @@ def run_python_style_checks(root_dir):
     run_cmd([os.path.join(root_dir, "dev", "lint-python")], stream_output=True)
 
 
+def run_pypi_packaging_tests(root_dir):
+    print("##### Running PyPi Packaging tests #####")
+
+    version = '0.0.0'
+    with open(os.path.join(root_dir, "version.sbt")) as fd:
+        version = fd.readline().split('"')[1]
+
+    # uninstall packages if they exist
+    run_cmd(["pip3", "uninstall", "--yes", "delta-spark", "pyspark"], stream_output=True)
+
+    print("### Clearing Delta artifacts from ivy2 and mvn cache")
+    run_cmd(["rm", "-rf", "~/.ivy2/cache/io.delta/"], stream_output=True)
+    run_cmd(["rm", "-rf", "~/.m2/repository/io/delta/"], stream_output=True)
+
+    install_cmd = ["pip3", "install", "wheel", "twine", "setuptools", "--upgrade"]
+    print("### Executing:" + " ".join(install_cmd))
+    run_cmd(install_cmd, stream_output=True)
+
+    dist_dir = path.join(root_dir, "dist")
+
+    print("### Deleting `dist` directory if it exists")
+    delete_if_exists(dist_dir)
+
+    gen_artifacts_cmd_1 = ["python3", "setup.py", "bdist_wheel"]
+    print("### Executing: " + " ".join(gen_artifacts_cmd_1))
+    run_cmd(gen_artifacts_cmd_1, stream_output=True, stderr=open('/dev/null', 'w'))
+
+    gen_artifacts_cmd_2 = ["python3", "setup.py", "sdist"]
+    print("### Executing: " + " ".join(gen_artifacts_cmd_2))
+    run_cmd(gen_artifacts_cmd_2, stream_output=True)
+
+    # we need, for example, 1.1.0_SNAPSHOT not 1.1.0-SNAPSHOT
+    version_formatted = version.replace("-", "_")
+    delta_whl_name = "delta_spark-" + version_formatted + "-py3-none-any.whl"
+
+    install_whl_cmd = ["pip3", "install", path.join(dist_dir, delta_whl_name)]
```

Comment:
does this automatically install pyspark as well??

Reply:
Yes. It will output "Successfully installed delta-spark-1.1.0-SNAPSHOT pyspark-3.1.2"